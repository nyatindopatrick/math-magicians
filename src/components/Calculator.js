import React, { useState } from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from '../logic/calculate';

export default function Calculator() {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = buttonName => {
    const value = calculate({ total, next, operation }, buttonName);
    if (value.next !== undefined) {
      setNext(value.next);
    }
    if (value.total !== undefined) {
      setTotal(value.total);
    }
    if (value.operation !== undefined) {
      setOperation(value.operation);
    }
  };

  return (
    <div className="main">
      <div className="calculator-main">
        <h2>Let&#39;s do some math!</h2>
        <div className="app">
          <Display data={total === null ? '0' : total.toString()} />
          <ButtonPanel handleClick={handleClick} />
        </div>
      </div>
    </div>
  );
}
