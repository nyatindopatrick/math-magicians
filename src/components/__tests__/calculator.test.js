import React from 'react';
import { shallow } from 'enzyme';

import Calculator from '../Calculator';

describe('Calculator', () => {
  let wrapper;

  it('should include do some math!', () => {
    wrapper = shallow(<Calculator />);
    expect(wrapper.find('h2').text()).toContain('do some math!');
  });
});
