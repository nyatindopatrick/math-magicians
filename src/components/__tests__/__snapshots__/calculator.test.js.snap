// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should include do some math! 1`] = `
<div
  className="main"
>
  <div
    className="calculator-main"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      className="app"
    >
      <div
        className="screen"
      >
        0
      </div>
      <div
        className="button-panel"
      >
        <div>
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="AC"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="+/-"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="%"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#f5913e",
                "width": "25%",
              }
            }
            type="button"
            value="รท"
          />
        </div>
        <div>
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="7"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="8"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="9"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#f5913e",
                "width": "25%",
              }
            }
            type="button"
            value="X"
          />
        </div>
        <div>
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="4"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="5"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="6"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#f5913e",
                "width": "25%",
              }
            }
            type="button"
            value="-"
          />
        </div>
        <div>
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="1"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="2"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="3"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#f5913e",
                "width": "25%",
              }
            }
            type="button"
            value="+"
          />
        </div>
        <div>
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "50%",
              }
            }
            type="button"
            value="0"
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "e0e0e0",
                "width": "25%",
              }
            }
            type="button"
            value="."
          />
          <input
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#f5913e",
                "width": "25%",
              }
            }
            type="button"
            value="="
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
